app/models/user.rb

```diff
@@ -138,6 +138,31 @@ def is_admin
     is_global_admin || community_user&.is_admin || false
   end
 
+  # Used by network profile: does this user have a profile on that other comm?
+  def has_profile_on(community_id)
+    cu = community_users.where(community_id: community_id).first
+    !cu&.user_id.nil?
+  end
+
+  def reputation_on(community_id)
+    cu = community_users.where(community_id: community_id).first
+    cu&.reputation
+  end
+
+  def post_count_on(community_id)
+    cu = community_users.where(community_id: community_id).first
+    cu&.post_count
```

Comment:
Optional: let's be extra sure and return `0`  by default to avoid accidental bugs with consumers of the API doing a safe navigation check. Has an additional benefit of preventing "type hopping" by providing a stable API contract to consumers ("always returns an integer", etc).